src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IEnumerableConverterFactoryHelpers.cs

```diff
@@ -139,6 +139,7 @@ public static bool IsImmutableEnumerableType(this Type type)
             }
         }
 
+        // TODO: https://github.com/dotnet/runtime/issues/38593.
```

Comment:
small NIT: We usually try to avoid checking in TODO comments on the code, any reason why we can' get to these two on the same PR? Are we blocked by a linker feature?

Reply:
Or just omit the TODO, that's what we have issues for.

Reply:
I've addressed the TODO in this PR. I was initially considering an alternative approach which might not have required the added `DynamicDependencyAttribute` usages, but I can explore this in the future since avoiding the attributes is not a priority right now.